iroh-gateway/src/core.rs

```diff
@@ -0,0 +1,412 @@
+use axum::{
+    body::{self, Body, BoxBody},
+    error_handling::HandleErrorLayer,
+    extract::{Extension, Path, Query},
+    http::{header::*, StatusCode},
+    response::IntoResponse,
+    routing::get,
+    BoxError, Router,
+};
+use serde::Deserialize;
+use std::{borrow::Cow, collections::HashMap, time::Duration};
+use tower::ServiceBuilder;
+
+use crate::{
+    client::{Client, Request},
+    config::Config,
+    constants::*,
+    error::GatewayError,
+    response::{GatewayResponse, ResponseFormat},
+};
+
+pub struct Core {
+    pub config: Config,
+    client: Client,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct GetParams {
+    format: Option<String>,
```

Comment:
It should be possible to acutally put in here `ResponseFormat`, but might need some work

Reply:
Hope it's fine to punt this for now. Put in a todo.